### ISSUE TITLE
Chrome Visual Issues

### ISSUE THREAD

**Issue by danielhickman** [google/material-design-lite#987]
I've only noticed the following issues with Chromium based browsers. I know Chrome is not the best at being pixel perfect or nice to scale to 4K, but thought you guys should take a look into it. *Any idea when Chrome is going to be visually pleasing???* It's not a WebKit issue, other WebKit browsers look beautiful, same with IE too.
- [ ] Multiple Line Tooltips are blurry, `-webkit-font-smoothing: antialiased;` doesn't help
![image](https://cloud.githubusercontent.com/assets/5341898/8637227/be0918ee-283b-11e5-966b-7933d1c1303e.png)
- [ ] Checkboxes that are checked have gaps
![image](https://cloud.githubusercontent.com/assets/5341898/8637265/86a64adc-283d-11e5-81c9-81710c4ded55.png)
- [ ] Waves expand a few pixels to the left after animation
![desktop 07 12 2015 - 02 17 37 03_2](https://cloud.githubusercontent.com/assets/5341898/8637316/34688354-2840-11e5-84b4-b6eadc73d535.gif)
- [ ] Waves on FABs don't animate most of this time
![desktop 07 12 2015 - 02 17 37 03_2_1_1](https://cloud.githubusercontent.com/assets/5341898/8637340/926d7fbc-2841-11e5-9fbe-d43c6ba9b7cd.gif)

MDL version: current one live at: getmdl.io
Browser: Chrome
Browser version: 43.0.2357.132 m (64-bit)
Operating system: Windows 8.1

**Comment by crutchcorn** [comment]
+1

**Comment by kybarg** [comment]
+1

**Comment by Garbee** [comment]
Are you using a 4k monitor? Have you also tested in a clean profile with no extensions?

I am unable to replicate any of these issues with my system which uses a fairly standard 1080p monitor.

**Comment by danielhickman** [comment]
I am using a 4K monitor, but they also appear with lower than HD monitors upon zooming. I shall test 1080p tonight. 
![screenshot_2015-07-12-07-29-34](https://cloud.githubusercontent.com/assets/5341898/8638190/ec894438-2867-11e5-9f23-66396615fec9.png)

**Comment by kybarg** [comment]
@danielhickman Is the last screen is made with zoom? Cause all browsers have some rendeirng issues while zooming. It is not considered a bug if  you can't see it at 100% zoom;

**Comment by danielhickman** [comment]
@kybarg Yes, it is, it's quite noticeable in 4K but not everyone has a 4K display to test with. Also... @Garbee My [1680 x 1050 display](http://support.hp.com/us-en/document/c01182749) results in the same issues. (The one liner tooltip was the issue this time though)
![img_20150712_103527012](https://cloud.githubusercontent.com/assets/5341898/8639057/13224840-2883-11e5-83d7-7fc06742ed0d.jpg)
![img_20150712_103400280](https://cloud.githubusercontent.com/assets/5341898/8639056/13214c06-2883-11e5-8193-04b7f147ccce.jpg)

**Comment by Garbee** [comment]
Ah, looks like you have Windows set to render at a magnification as well. That is most likely the source of the oddities... Not sure how we can program to take this into account. Looks like it is just upstream issues with how Chrome interacts with the magnification done by the OS.

**Comment by danielhickman** [comment]
I guess that makes sense. Is there an easy way to make sure this gets to Chrome devs? Every other browser I've used works perfectly with this.

**Comment by addyosmani** [comment]
> I guess that makes sense. Is there an easy way to make sure this gets to Chrome devs? 

We're here! Would it be possible to share what system magnification settings (if using them) you've got setup atm? Any further info that can help us reproduce this would be great, though if this is a Chrome rendering bug we'll want to get this filed at crbug.com.

> Every other browser I've used works perfectly with this.

Although we've got a relatively decent device lab setup, I don't yet have a 4K screen I'm afraid :) I do have Windows machines and could possibly emulate what you're seeing but like @Garbee I'm primarily testing on 1080p machines.

**Comment by S-YOU** [comment]
If it is about font anti-aliasing, I think there is one flag that has been changed in chrome:flags since chrome 4x.x

**Comment by kybarg** [comment]
@S-YOU you are may be right.
@danielhickman try to go <code>chrome://flags</code> and reset to defaults.

**Comment by S-YOU** [comment]
As far as I remembered, it is new default by chrome, may be font directwrite setting. Windows only option, I don't have Windows at the moment, to double check it.

**Comment by danielhickman** [comment]
**Setup in Windows:**

![05_18_45](https://cloud.githubusercontent.com/assets/5341898/8649412/e61f92f2-291e-11e5-809d-0ce2068d402d.png)
![05_18_28](https://cloud.githubusercontent.com/assets/5341898/8649413/e6321c10-291e-11e5-9c4d-1102b91fb377.png)

**Monitor:**

[Link](http://www.ncixus.com/products/?sku=95635)

I Reset all Chrome Flags to their defaults, restarted Chrome, and all bugs are repeatable still. The tooltips are fuzzy only after "Empty Cache and Hard Reload" (Access by right clicking the reload button in the development console) but then didn't work the next time I tried. I'll finish up a few things I need to do and I'll try testing with the resolutions and scaling. :smile:

**Comment by danielhickman** [comment]
### 1080p with the default scaling
**Perfect, even at 400% zoom**

![06_04_41](https://cloud.githubusercontent.com/assets/5341898/8650106/26cb84fe-2925-11e5-9ee5-329cd55e7840.png)

### 1080p with the 'larger' (150%) scaling
**Same issue, even without zooming**

![06_01_34](https://cloud.githubusercontent.com/assets/5341898/8650107/26dbcc24-2925-11e5-8f38-a47e0495e8dd.png)

**Comment by Garbee** [comment]
Yea, so it is related to the OS scaling. Thanks for testing that more and reporting back!

I don't think there is much MDL can do directly to address this. Seems like something we need to upstream to Chromium and get the engineers there to look at.

@addyosmani Sound good sending this up to [crbug](http://crbug.com) now? Or do you think there is something actionable for us here?

**Comment by danielhickman** [comment]
Looks like the same is happening with Polymer too.

**Comment by Garbee** [comment]
If it happens with Polymer as well, then you need to file a bug upstream against Chromium to be addressed. Nothing we can do here to check for OS magnification and do something differently based on it (plus that would be hell anyways.)

**Comment by danielhickman** [comment]
Any news on this? I know this is still in the wrong place. I haven't seen any changes on any of my devices. I notice it everywhere so it might be an issue with the rendering engine? (I have no clue).

**Comment by willzhang05** [comment]
I've also noticed a Chrome (and Safari) visual issue (But it doesn't appear on Edge or Firefox), and I'm not exactly sure what's causing it. It occurs both on http://getmdl.io/components/index.html#buttons-section and my test page. It seems that when the button ripple changes states, the border-radius of the button becomes 0 somehow and the button isn't rounded for a brief second before becoming rounded again.

**Screenshot of getmdl.io website:**
![ripple](https://cloud.githubusercontent.com/assets/8996428/13098102/0c53dd52-d4f5-11e5-85de-278868f1d237.png)

**Screenshot of test page:**
![Image](http://i.imgur.com/hEYA9ia.png)

**Comment by sgomes** [comment]
Most of these issues should be fixed by now. Please file individual new issues if there's anything that slipped between the cracks.
